CIPs/CIP-124/cip-124.md

```diff
@@ -0,0 +1,321 @@
+    - Can we quickly disseminate new EventIDs to interested nodes?
+- Low overhead for the querying peer (e.g. no requirement to connect to and query a massive amount of peers)
+    - Can we keep most nodes up to date so point queries can go to a small number of, or single, node(s)?
+
+By pulling some set of streams out of the main network channel into a stream set the nodes interested in those streams can synchronies with each other without putting load on the uninterested nodes in the network. These Stream Sets will likely be defined by a set of models that should be included in the Stream Set. They could also be defined by a set of models and controllers but the large and changing number of controllers requires more though and is not proposed here.
+
+A stream also could potentially be included in more than one stream set if for example there are apps that want to sync every event across the entire network, they may want to form a stream set for all events that overlaps the narrower stream sets.
+
+The introduction of stream sets should lower the burden of running a node that is not interested in a stream set and speed up the process of syncing all the historical events that are within the stream set.
+
+A secondary goal of stream sets is that they give a structure for sharding the stream set across ceramic nodes. By supporting an ability to synchronize a sub-range of the stream set the load of storing, indexing, and retrieving streams can be split among ceramic nodes.
+
+Ceramic nodes also need to have a way to find the other nodes interested in the stream set so that they can synchronize with them.
+
+Ultimately by gossiping about ranges of streams ceramic nodes that are arbitrarily out of sync can synchronize all events of mutual interest. Nodes that are in sync or nearly in sync can send each other very little bandwidth. Nodes can avoid sending stream event announcements to nodes that have no interest in the streams.
+
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+Specification goes here.
+
+### **Stream Sets Ranges (Ordering)**
+We need to choose a consistent ordering to give ranges meaning.
+What order should we be sorting the EventIDs
+`/ceramic/mainnet/modal_family/modal/stream/event_hight/event`
+`/ceramic/mainnet/anchor_time/event(64)`
+
+#### **EventID sort order**
+Should we also propose a CIP to change EventIDs so that they sort better lexicographically
+> 🚧 The EventIDs are grouped by stream but then in random order
+```tsx
+// Curent EventID
+<EventID> ::= <multibase-prefix><multicodec-streamid><stream-type>
+  <genesis-cid-bytes>
+  <event-cid-bytes>
+
+// Proposed EventID
+<EventID> ::= <multibase-prefix><multicodec-streamid><stream-type>
+  <genesis-cid-bytes>
+  <multicodec-unsigned-varint><event hight>
+  <event-cid-bytes>
+```
+[multiformats/unsigned-varint](https://github.com/multiformats/unsigned-varint)
```

Comment:
Actually something that could make even more sense and would be less ambiguous:

```xml
<EventID> ::= <multibase-prefix><multicodec-streamid><stream-type><event-info>

<event-info> ::= <multicodec-eventid><truncated-streamid-bytes><varint event-height><event-cid>
```

This would allow us to more easily distinguish if it's an init event (just a CID multicodec) or event-id (with it's own multicodec)